common/evmcompat/helpers.go

```diff
@@ -191,3 +191,85 @@ func SolidityPackedBytes(pairs []*Pair) ([]byte, error) {
 
 	return b.Bytes(), nil
 }
+
+func SolidityUnpackString(data string, types []string) ([]interface{}, error) {
+
+	if data[0:2] == "0x" {
+		data = data[2:]
+	}
+	var resp = make([]interface{}, len(types))
+	var stringCount = 0
+	for i := 0; i < len(types); i++ {
+		partialData := data[i*64 : (i+1)*64]
+		convertedData, isString, err := TypeConverter(partialData, types[i], data[i*64:], len(types)-i, stringCount)
+		if err != nil {
+			return nil, err
+		}
+		if isString {
+			stringCount += 1
+		}
+		resp[i] = convertedData
+	}
+	return resp, nil
+}
+
+func TypeConverter(partialData, typeString, dataLeft string, chunkLeft, stringCount int) (interface{}, bool, error) {
```

Comment:
`TypeConverter` -> `parseNextValueFromSolidityHexStr`
This function doesn't need to be exported.